Firmware/Marlin_main.cpp

```diff
@@ -1040,15 +1048,13 @@ void setup()
 	if ((farm_mode == 0xFF && farm_no == 0) || ((uint16_t)farm_no == 0xFFFF)) 
 		farm_mode = false; //if farm_mode has not been stored to eeprom yet and farm number is set to zero or EEPROM is fresh, deactivate farm mode
 	if ((uint16_t)farm_no == 0xFFFF) farm_no = 0;
-	
-	selectedSerialPort = eeprom_read_byte((uint8_t*)EEPROM_SECOND_SERIAL_ACTIVE);
-	if (selectedSerialPort == 0xFF) selectedSerialPort = 0;
 	if (farm_mode)
 	{
 		no_response = true; //we need confirmation by recieving PRUSA thx
 		important_status = 8;
 		prusa_statistics(8);
 		selectedSerialPort = 1;
+		MYSERIAL.begin(BAUDRATE);
```

Comment:
why is the serial being initialized again? See line 1020

Reply:
Yes. If selectedSerialPort was disabled prior to the farm_mode check, then the serial would be broken on the first farm mode boot

Reply:
When the serial was initialized later in the code this wasn't an issue, but now that we initialize it so early we really need this extra re-init. In any case, it just sets the UART registers again. No buffer manipulation happens.